src/java.net.http/share/classes/jdk/internal/net/http/CheckedIterable.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package jdk.internal.net.http;
+
+import java.util.Iterator;
+
+/**
+ * An {@link Iterable} clone supporting checked exceptions.
+ *
+ * @param <E> the type of elements returned by the produced iterators
+ */
+@FunctionalInterface
+interface CheckedIterable<E> {
+
+    /**
+     * {@return an {@linkplain CheckedIterator iterator} over elements of type {@code E}}
+     */
+    CheckedIterator<E> iterator() throws Exception;
+
+    static <E> CheckedIterable<E> fromIterable(Iterable<E> iterable) {
+        return () -> {
+            Iterator<E> iterator = iterable.iterator();
+            return CheckedIterator.fromIterator(iterator);
+        };
```

Comment:
```suggestion
        return () -> CheckedIterator.fromIterator(iterable.iterator());
```

Reply:
I will keep the current style which enables stepping convenience in the debugger.